### ISSUE TITLE
No Online Data Collection

### ISSUE THREAD

**Issue by albertwilcox** [hari-sikchi/AWAC#2]
Hi,

Thanks for this repo! The offline learning seems to work great. However, I noticed that although there are comments referencing online data collection and the awac object stores and online exploration specific environment, there is no actual code to produce or save online data.

**Comment by albertwilcox** [comment]
I'm currently implementing it so if you want me to submit a pull request lmk

**Comment by hari-sikchi** [comment]
Hi Albert! Thanks for your interest in the repo. Yes, a pull request for online data collection would be a great addition. In my experience, I found that AWAC with online data collection has some issues trying to reach asymptotic performance. One possibility might be its KL constraint to replay buffer which might contain bad transitions.
Let me know if you face anything similar. Thanks!

**Comment by marianophielipp** [comment]
Hi Hari

 This seems a nice implementation. But it was not clear to me in your code how are checking the update every n number of offline steps. I see that you did not use the time step parameter inside the update function. Also was not clear to me how you used the advantage \ train function needed for Eq 10 of the paper. Thank you. Best regards, Mariano.

**Comment by hari-sikchi** [comment]
Hi Mariano,
1. The code currently does not extend to online training. It just uses offline data and gives the best near-replay-buffer-distribution policy. So, we just keep on updating every single timestep and evaluate after some fixed timesteps. For online training, I have found some issues which I have commented on in the issue thread.
2. Yes, the update time-step parameter is not used and is redundant. 
3. The advantage function, if used as given in equation 10 results in unstable training as the advantage can be quite large at times, so the advantages are batch normalized as seen in the 'weights' parameter. This makes the gradient biased but works well in practice.

**Comment by marianophielipp** [comment]
Hi @albertwilcox any ETA on the online PR?

**Comment by marianophielipp** [comment]
Thank you @hari-sikchi for the clarification!

**Comment by albertwilcox** [comment]
@marianophielipp Hi, I don't have the time to run extensive tests but I can push something later today. I got it to work about as well as the paper on Ant but haven't tested it on anything else
